docs(adk): rewrite Managing Integrations page (#451)
botpress/docs: adk/managing-integrations.mdx

---

> title: Managing Integrations

Reviewer: ```suggestion
title: Managing integrations
```

titles and headings should use [sentence case](https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings)

Author: Changed! Thanks for pointing it out!

---

Integrations connect your agent to external services and platforms. This guide covers how to find, add, configure, and manage integrations in your ADK project.

## Finding integrations

### Search the hub

Search for integrations by keyword:

```bash
adk search crm
```

### List available integrations

> 

Reviewer: **suggestion**: link to the reference for these commands so users can quickly get to any flags

Author: added!

Browse all available integrations from the Botpress hub:

```bash
adk list --available
```

### View integration details

Get detailed information about a specific integration, including its actions, events, and channels:

```bash
adk info linear
```

You can also filter the output:

```bash
adk info linear --actions    # Show only actions
adk info linear --events     # Show only events
adk info linear --channels   # Show only channels
```

## Adding integrations

Add an integration using the `adk add` command:

```bash
adk add linear
```

```txt
✓ Added linear version 2.1.2 as linear to agent.config.ts dependencies

Added as disabled. Enable and configure it in the Control Panel.
```

This adds the integration to your `agent.config.ts` as a string shorthand:

```typescript
dependencies: {
  integrations: {
    chat: "chat@0.7.7",
    webchat: "webchat@0.3.0",
    linear: "linear@2.1.2",
  },
},
```

<Note>

>   Integrations are added as disabled by default. Enable and configure them in the Control Panel during `adk dev`.

Reviewer: ```suggestion
  Newly added integrations are disabled by default. Enable and configure them in the Control Panel during `adk dev`.
```

slightly awkward wording, this should be fixed in the CLI output as well

Author: Yes, changed!

</Note>

### Pinning a version

Add a specific version instead of the latest:

```bash
adk add notion@3.0.0
```

### Using an alias

> 

Reviewer: might be worth mentioning that you can't alias the Webchat integration, up to you

Author: Very nice observations! Added a note to this

Add an integration with a custom alias. This is useful when you need multiple instances of the same integration:

```bash
adk add webhook --alias webhook-orders
```

The alias is how you reference the integration in your code:

```typescript
import { actions } from "@botpress/runtime";

await actions["webhook-orders"].send({ url: "https://api.example.com/orders", data: payload });
```

### Workspace-scoped integrations

Add a private or custom integration from your workspace:

```bash
adk add my-workspace/custom-integration@1.0.0
```

## Configuring integrations

Integration settings (API keys, tokens, OAuth credentials) are managed in the Control Panel, not in `agent.config.ts`.

1. Run `adk dev` to start the development server
2. Open the Control Panel at `http://localhost:3001/integrations`
3. Find the integration and click to configure it
4. Enter the required settings and enable it

> <Frame>

Reviewer: consult the [README section about using images](https://github.com/botpress/docs?tab=readme-ov-file#images). if they're UI screenshots there should be both a light and dark version

Author: added light and dark mode screenshots

───────────────────────────────────────────────────────────────
chat                chat                          0.7.7
webchat             webchat                       0.3.0
linear              linear                        2.1.2
```

## Updating integrations

### Update a specific integration

```bash
adk upgrade linear
```

### Update all integrations interactively

```bash
adk upgrade
```

This shows available updates and lets you choose which to upgrade:

```txt
✓ Upgraded:
  • linear (2.0.0 → 2.1.2)

Run adk dev to install the updated integrations
```

## Removing integrations

Remove an integration from your project:

```bash
adk remove linear
```

```txt
✓ Removed linear from agent.config.ts dependencies
  Run adk dev to update your project
```

You can also run `adk remove` without a name to choose interactively.

## Using integrations in code

Once an integration is installed and enabled, its actions become available in your agent:

```typescript
import { actions } from "@botpress/runtime";

// Call an integration action
await actions.linear.createIssue({
  title: "Bug report from user",
  teamId: "TEAM-123",
});
```

> Integration events can be subscribed to in [Triggers](/adk/concepts/triggers) and [Conversations](/adk/concepts/conversations) (via the `events` prop).

Reviewer: ```suggestion
You can subscribe to integration events in [Triggers](/adk/concepts/triggers) and [Conversations](/adk/concepts/conversations) (via the `events` prop).
```

passive voice makes this sentence a bit hard to parse

Author: Updated! Will keep this in mind in the future

```bash
adk info linear
```

You can also filter the output:

```bash
adk info linear --actions    # Show only actions
adk info linear --events     # Show only events
adk info linear --channels   # Show only channels
```

See the [CLI reference](/adk/cli-reference) for all available flags and options.

## Adding integrations

Add an integration using the `adk add` command:

```bash
adk add linear
```

```txt
✓ Added linear version 2.1.2 as linear to agent.config.ts dependencies

Added as disabled. Enable and configure it in the Control Panel.
```

This adds the integration to your `agent.config.ts` as a string shorthand:

```typescript
dependencies: {
  integrations: {
    chat: "chat@0.7.7",
    webchat: "webchat@0.3.0",
    linear: "linear@2.1.2",
  },
},
```

<Note>
  Newly added integrations are disabled by default. Enable and configure them in the Control Panel during `adk dev`.
</Note>

### Pinning a version

Add a specific version instead of the latest:

```bash
adk add notion@3.0.0
```

### Using an alias

Add an integration with a custom alias. This is useful when you need multiple instances of the same integration:

```bash

> adk add webhook --alias webhook-orders

Reviewer 2: **nit:** webhook-orders is not a valid identifier unless it's in quotes. I'm about to add a PR that doesn't wrap the integrations keys in quotes. They should be valid typescript identifiers, so hyphenated keys will throw an error.

Author: good point! changed!

  integrations: {
    chat: "chat@0.7.7",
    webchat: "webchat@0.3.0",
    linear: "linear@2.1.2",
  },
},
```

<Note>
  Newly added integrations are disabled by default. Enable and configure them in the Control Panel during `adk dev`.
</Note>

### Pinning a version

Add a specific version instead of the latest:

```bash
adk add notion@3.0.0
```

### Using an alias

Add an integration with a custom alias. This is useful when you need multiple instances of the same integration:

```bash
adk add webhook --alias webhook-orders
```

The alias is how you reference the integration in your code:

```typescript
import { actions } from "@botpress/runtime";

await actions["webhook-orders"].send({ url: "https://api.example.com/orders", data: payload });
```

<Note>
  Avoid aliasing built-in channel integrations (webchat, chat, slack, teams, telegram, whatsapp). The runtime uses the alias to identify these integrations, and a custom alias will prevent default message components from loading.
</Note>

### Workspace-scoped integrations

Add a private or custom integration from your workspace:

```bash
adk add my-workspace/custom-integration@1.0.0
```

## Configuring integrations

> Integration settings (API keys, tokens, OAuth credentials) are managed in the Control Panel, not in `agent.config.ts`.

Reviewer 2: suggestion: I think it's worth it to explicitly mention that this is behaviour that changed from before and add a walkthrough of the migration steps in the CLI (the modal that pops up and asks users to switch to the new format)

1. Run `adk dev` to start the development server
2. Open the Control Panel at `http://localhost:3001/integrations`
3. Find the integration and click to configure it
4. Enter the required settings and enable it

<Frame>
  <img alt="Integrations in the Control Panel" src="./assets/adk-integration-page.png" />
</Frame>

Each integration card in the Control Panel shows its current status:

| Status | Meaning |
|--------|---------|
| Connected | Enabled, registered, and configured |
| Disabled | Turned off, can be enabled |
| Needs Action | Requires installation, configuration, or enabling |
| Failed | Registration error |

## Listing installed integrations

View integrations currently installed in your project:

```bash
adk list
```

```txt
Installed Integrations (3)

Alias               Name                          Version
───────────────────────────────────────────────────────────────
chat                chat                          0.7.7
webchat             webchat                       0.3.0
linear              linear                        2.1.2
```

## Updating integrations

### Update a specific integration

```bash

> adk upgrade linear

Reviewer 2: suggestion: make it clear that this is keyed by the alias not the name of the integration on the hub.

Author: yes
